src/order-book/api.ts

```diff
@@ -51,43 +58,132 @@ function cleanObjectFromUndefinedValues(obj: Record<string, string>): typeof obj
   }, {} as typeof obj)
 }
 
+/**
+ * The parameters for the `getOrders` request.
+ */
+export type GetOrdersRequest = {
+  owner: Address
+  offset?: number
+  limit?: number
+}
+
+/**
+ * The CoW Protocol OrderBook API client.
+ *
+ * This is the main entry point for interacting with the CoW Protocol OrderBook API. The main advantage of using
+ * this client is the batteries-included approach to interacting with the API. It handles:
+ *
+ * - Environment configuration (mainnet, staging, etc.)
+ * - Rate limiting
+ * - Retries
+ * - Backoff
+ * - Error handling
+ * - Request signing
+ * - Request validation
+ *
+ * @example
+ *
+ * ```typescript
+ * import { OrderBookApi, OrderSigningUtils, SupportedChainId } from '@cowprotocol/cow-sdk'
+ * import { Web3Provider } from '@ethersproject/providers'
+ *
+ * const account = 'YOUR_WALLET_ADDRESS'
+ * const chainId = 5 // Goerli
+ * const provider = new Web3Provider(window.ethereum)
+ * const signer = provider.getSigner()
+ *
+ * const quoteRequest = {
+ *   sellToken: '0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6', // WETH goerli
+ *   buyToken: '0x02abbdbaaa7b1bb64b5c878f7ac17f8dda169532', // GNO goerli
+ *   from: account,
+ *   receiver: account,
+ *   sellAmountBeforeFee: (0.4 * 10 ** 18).toString(), // 0.4 WETH
+ *   kind: OrderQuoteSide.kind.SELL,
+ * }
+ *
+ * const orderBookApi = new OrderBookApi({ chainId: SupportedChainId.GOERLI })
+ *
+ * async function main() {
+ *     const { quote } = await orderBookApi.getQuote(quoteRequest)
+ *
+ *     const orderSigningResult = await OrderSigningUtils.signOrder(quote, chainId, signer)
+ *
+ *     const orderId = await orderBookApi.sendOrder({ ...quote, ...orderSigningResult })
+ *
+ *     const order = await orderBookApi.getOrder(orderId)
+ *
+ *     const trades = await orderBookApi.getTrades({ orderId })
+ *
+ *     const orderCancellationSigningResult = await OrderSigningUtils.signOrderCancellations([orderId], chainId, signer)
+ *
+ *     const cancellationResult = await orderBookApi.sendSignedOrderCancellations({...orderCancellationSigningResult, orderUids: [orderId] })
+ *
+ *     console.log('Results: ', { orderId, order, trades, orderCancellationSigningResult, cancellationResult })
+ * }
```

Comment:
Great idea!
did you check if this example works out of the box?

Reply:
This is one that I haven't checked actually. The 🦀 within would appreciate it if JavaScript could do doc testing like rust 😅

Reply:
Alternativelly we can point to the docs and make sure it works https://docs.cow.fi/cow-sdk/cow-api

Reply:
This is being tracked in #139 for completion once these series of PRs are done.